sql/src/main/java/io/crate/metadata/cluster/RenameIndexClusterStateExecutor.java

```diff
@@ -22,69 +22,54 @@
 
-    public RenameIndexClusterStateExecutor(Settings settings,
-                                           IndexNameExpressionResolver indexNameExpressionResolver) {
+    public RenameIndexClusterStateExecutor() {
         logger = LogManager.getLogger(getClass());
-
-        this.indexNameExpressionResolver = indexNameExpressionResolver;
     }
 
     @Override
-    public ClusterState execute(ClusterState currentState, RenameIndexRequest request) throws Exception {
-        final String[] sourceIndexName = request.sourceIndexName();
-        final String[] targetIndexName = request.targetIndexName();
+    public ClusterState execute(ClusterState currentState, BulkRenameIndexRequest request) throws Exception {
 
-        logger.info("rename indexes '[{}]' to '[{}]'", sourceIndexName, targetIndexName);
+        final String[] sourceIndexNames = request.sourceIndices();
+        final String[] targetIndexNames = request.targetIndices();
 
+        logger.info("rename indexes '[{}]' to '[{}]'", sourceIndexNames, targetIndexNames);
         MetaData.Builder mdBuilder = null;
         ClusterBlocks.Builder blocksBuilder = ClusterBlocks.builder()
             .blocks(currentState.blocks());
 
-        Index[] sourceIndices = indexNameExpressionResolver.concreteIndices(currentState,
-            STRICT_INDICES_OPTIONS, sourceIndexName);
-        Index[] sourceIndicesOrdered = new Index[sourceIndexName.length];
-
-        // ensure the order in input is respected
-        for (int i = 0; i < sourceIndexName.length; i++) {
-            for (int j = 0; j < sourceIndices.length; j++) {
-                if (sourceIndexName[i].equals(sourceIndices[j].getName())) {
-                    sourceIndicesOrdered[i] = sourceIndices[j];
-                    break;
-                }
-            }
-        }
-
         MetaData metaData = currentState.getMetaData();
         mdBuilder = MetaData.builder(metaData);
 
-        renameIndices(metaData, mdBuilder, blocksBuilder, sourceIndicesOrdered, targetIndexName);
+        renameIndices(metaData, mdBuilder, blocksBuilder, sourceIndexNames, targetIndexNames);
+        return ClusterState.builder(currentState).metaData(mdBuilder).blocks(blocksBuilder).build();
+    }
 
-        // The MetaData will always be overridden (and not merged!) when applying it on a cluster state builder.
-        // So we must re-build the state with the latest modifications before we pass this state to possible modifiers
-        // or return it.
-        // Otherwise they would operate on the old MetaData and would just ignore any modifications.
-        currentState = ClusterState.builder(currentState).metaData(mdBuilder).blocks(blocksBuilder).build();
-        return currentState;
+    static void renameIndices(MetaData metaData,
+                              MetaData.Builder mdBuilder,
+                              ClusterBlocks.Builder blocksBuilder,
+                              String[] sourceIndexNames,
```

Comment:
Could we use the new the new RenameAction structure here? Otherwise we should at least add some basic sanity checks - that the arrays have the same size.

Reply:
done